landoapi/decorators.py

```diff
@@ -0,0 +1,90 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import functools
+import logging
+
+from connexion import problem
+from flask import request
+from netaddr import IPAddress, IPNetwork
+
+logger = logging.getLogger(__name__)
+
+
+def get_ips():
+    if len(request.access_route) > 0:
+        return list(request.access_route)
+    return [request.remote_addr]
+
+
+def limit_ips(
+    whitelist_ip,
+    inbound_proxies,
+    log_tag,
+    log_msg='Access denied',
+    err_title='Not authorized',
+    err_msg='Access denied'
+):
+    """Allow only whitelisted IPs.
+
+    Args:
+        whitelist_ip: list
+            allowed Networks ['127.0.0.1', '192.168.0.1/8']
+        inbound_proxies: list
+            allowed proxies IP addresses
+        log_tag: string
+            tag the warning in logs
+        log_msg: string
+            Message to log
+        err_title: string
+            Title of the message to respond to the client
+        err_message: string
+            Body of the message to respond to the client
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+            ips = get_ips()
+            ip = ips.pop(0)
+
+            valid_proxy = False
+            if (len(ips) > 0 and not set(inbound_proxies) & set(ips)):
+                logger.warning(
+                    {
+                        'remote_addr': ip,
+                        'msg': 'Wrong proxy. %s' % log_msg,
+                    }, log_tag
+                )
+                return problem(
+                    403,
+                    err_title,
+                    err_msg,
+                    type=
+                    'https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403'
+                )
+
+            in_whitelist = False
+            for ip_mask in whitelist_ip:
+                if IPAddress(ip) in IPNetwork(ip_mask):
+                    in_whitelist = True
+                    break
+            if not in_whitelist:
+                logger.warning(
+                    {
+                        'remote_addr': ip,
+                        'msg': 'Wrong IP address. %s' % log_msg,
```

Comment:
"wrong ip address" isn't a useful error message.

try something along the lines of "the ip 0.0.0.0 is not allowed to connect"